common.sh

```diff
@@ -126,6 +123,19 @@ function fetch_tools {
     curl -sL ${kb_tools_download_url} -o "$kb_tools_archive_path"
   fi
   tar -zvxf "$kb_tools_archive_path" -C "$tmp_root/"
+
+  header_text "ensuring dep is present for legacy v1 scaffold tests"
+  if ! which dep; then
+    mkdir -p test_gobin
+    # can't install dep with modules due to dependency versioning (lol)
+    curl -sL -o ${test_gobin}/dep https://github.com/golang/dep/releases/download/0.5.2/dep-$(go env GOOS)-$(go env GOARCH)
```

Comment:
I get the file but it's not executable, `chmod +x` please?
PS in github I'm not sure if I should use "Add a single comment" or "Start a review", I'm picking at random

Reply:
I'm testing this stuff so I can see what happens using `bash -exu ./test.sh` and I found that the `mkdir -p test_gobin` isn't using the variable `${test_gobin}`.  I know that I had dep in the /tmp/.. directory so it's probably created elsewhere and not failing here

Reply:
What about incorporating the `dep` into the dependency graph as a [build-scoped tool](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module)?

Reply:
dep, unfortunately, doesn't install properly using using modules without quite a bit of fiddling (mismatched versions, somewhat ironically).